### ISSUE TITLE
Instruction guidance?

### ISSUE THREAD

**Issue by oaustegard** [HKUNLP/instructor-embedding#59]
While the paper is insightful and the performance speaks for itself it is not at all clear to me what the guidelines are around constructing the instructions for the embedding fine-tuning.  It would be beneficial to provide some additional examples and general how-tos around this.  E.g. Are the prompts here similar to what one might apply to an LLM?  Is there a token-limit on the prompt? Are there specific words/complete prompts that elicit specific behavior?  How can we test one instruction's performance vs another? etc, etc

**Comment by hongjin-su** [comment]
Hi, we have a unified instruction template in the section 2.3 of our paper. With INSTRUCTOR model, we have default maximum length 512, and may not have additional token-limit for the instruction. We provide some examples for the words in  table 4 of our paper. To test the instruction's performance, you may have an validation set for evaluation.

**Comment by oaustegard** [comment]
Right, I should have been more specific:  
so the general recommended structure is “REPRESENT THE (**DOMAIN**) **TEXT TYPE** FOR **TASK OBJECTIVE**"

How does altering the DOMAIN, TEXT TYPE and TASK/OBJECTIVE alter the behavior of the embedder?  And how does the user know what values are valid/effective?

I guess my question boils down to: _how would a user readily iterate the instructions/prompts to arrive at the best prompt for their domain?_

As you state:
At the core of INSTRUCTOR is the instruction design. While our current unified instruction format has demonstrated effectiveness, future research can explore other instructional elements to further improve performance.

Any pointers in how that can be achieved?

**Comment by hongjin-su** [comment]
Hi, the instruction words are usually obtained from the texts to be encoded. Users may determine the domain, text type and purpose of the embedding texts. With several equally good candidates, they may build a small validation set to perform visualization or benchmark scores

**Comment by hongjin-su** [comment]
Please re-open the issue if you have any questions or comments!
